drivers/gpio/egpio_nrfx.c

```diff
@@ -0,0 +1,194 @@
+
+	int ret_val = ipc_service_send(ep, (void*)msg, sizeof(nrfe_gpio_data_packet));
+
+	if (ret_val != 0) {
+		return ret_val;
+	}
+
+	struct nrfe_gpio_data_packet msg = {.opcode = NRFE_GPIO_PIN_CLEAR,
+									.pin = clear_mask,
+									.port = get_port_cfg(port)->port_num};
+
+	return ipc_service_send(ep, (void*)msg, sizeof(nrfe_gpio_data_packet));
+}
+
+static int gpio_nrfx_port_set_bits_raw(const struct device *port,
+				       gpio_port_pins_t mask)
+{
+	struct nrfe_gpio_data_packet msg = {.opcode = NRFE_GPIO_PIN_SET,
+									.pin = mask,
+									.port = get_port_cfg(port)->port_num};
+
+	return ipc_service_send(ep, (void*)msg, sizeof(nrfe_gpio_data_packet));
+}
+
+static int gpio_nrfx_port_clear_bits_raw(const struct device *port,
+					 gpio_port_pins_t mask)
+{
+	struct nrfe_gpio_data_packet msg = {.opcode = NRFE_GPIO_PIN_CLEAR,
+									.pin = mask,
+									.port = get_port_cfg(port)->port_num};
+
+	return ipc_service_send(ep, (void*)msg, sizeof(nrfe_gpio_data_packet));
+}
+
+static int gpio_nrfx_port_toggle_bits(const struct device *port,
+				      gpio_port_pins_t mask)
+{
+	struct nrfe_gpio_data_packet msg = {.opcode = NRFE_GPIO_PIN_TOGGLE,
+									.pin = mask,
+									.port = get_port_cfg(port)->port_num};
+
+	return ipc_service_send(ep, (void*)msg, sizeof(nrfe_gpio_data_packet));
+}
+
+static int gpio_nrfx_init(const struct device *port)
+{
+	const struct device *ipc0_instance = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+
+	ret = ipc_service_open_instance(ipc0_instance);
+	if ((ret < 0) && (ret != -EALREADY)) {
+		LOG_ERR("ipc_service_open_instance() failure");
+		return ret;
+	}
+
+	ret = ipc_service_register_endpoint(ipc0_instance, &ep, &ep_cfg);
+	if (ret < 0) {
+		LOG_ERR("ipc_service_register_endpoint() failure");
+		return ret;
+	}
+
+	k_sem_take(&bound_sem, K_FOREVER);
+
+	return 0;
+}
+
+static const struct gpio_driver_api gpio_nrfx_drv_api_funcs = {
+	.pin_configure = gpio_nrfx_pin_configure,
+	.port_set_masked_raw = gpio_nrfx_port_set_masked_raw,
+	.port_set_bits_raw = gpio_nrfx_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_nrfx_port_clear_bits_raw,
+	.port_toggle_bits = gpio_nrfx_port_toggle_bits,
+};
+
+#define GPIOTE_PHANDLE(id) DT_INST_PHANDLE(id, gpiote_instance)
+#define GPIOTE_INST(id)    DT_PROP(GPIOTE_PHANDLE(id), instance)
+
+#define GPIOTE_INSTANCE(id)                                     \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(id, gpiote_instance), \
+		    (NRFX_GPIOTE_INSTANCE(GPIOTE_INST(id))),    \
+		    ({ .p_reg = NULL }))
+
+/* Device instantiation is done with node labels because 'port_num' is
+ * the peripheral number by SoC numbering. We therefore cannot use
+ * DT_INST APIs here without wider changes.
+ */
+
+#define GPIOTE_CHECK(id)						       \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(id, gpiote_instance),		       \
+		(BUILD_ASSERT(DT_NODE_HAS_STATUS(GPIOTE_PHANDLE(id), okay),    \
+			"Please enable GPIOTE instance for used GPIO port!")), \
+		())
```

Comment:
I don't think this driver should use GPIOTE nodes.
The only part that should be allocated for it are the pins.
Perhaps a new binding is needed?